src/bin/lttng-sessiond/kernel.c

```diff
@@ -191,6 +193,227 @@ int kernel_create_channel(struct ltt_kernel_session *session,
+			LTTNG_USERSPACE_PROBE_LOCATION_LOOKUP_METHOD_TYPE_TRACEPOINT_SDT);
+
+
+	probe_name = lttng_userspace_probe_location_tracepoint_get_probe_name(
+			probe_location);
+	if (!probe_name) {
+		ret = -1;
+		goto end;
+	}
+
+	provider_name = lttng_userspace_probe_location_tracepoint_get_provider_name(
+			probe_location);
+	if (!provider_name) {
+		ret = -1;
+		goto end;
+	}
+
+	fd = lttng_userspace_probe_location_tracepoint_get_binary_fd(probe_location);
+	if (fd < 0) {
+		ret = -1;
+		goto end;
+	}
+
+	ret = run_as_extract_sdt_probe_offsets(fd, provider_name, probe_name,
+			session->uid, session->gid, offsets, num_offset);
+	if (ret < 0) {
+		DBG("userspace probe offset calculation failed for sdt "
+			"probe %s:%s", provider_name, probe_name);
+		goto end;
+	}
+
+	for (int i = 0; i < *num_offset; i++) {
+		DBG("userspace probe sdt offsets found for %s:%s at 0x%lx.",
+				provider_name, probe_name, (*offsets)[i]);
+	}
+end:
+	return ret;
+}
+
+/*
+ * Extract the offsets of the instrumentation point for the different lookup
+ * methods.
+ */
+static
+int userspace_probe_add_callsites(struct lttng_event *ev,
+			struct ltt_kernel_session *session, int fd)
+{
+	int ret;
+
+	struct lttng_userspace_probe_location *location = NULL;
+	struct lttng_userspace_probe_location_lookup_method *lookup_method = NULL;
+	enum lttng_userspace_probe_location_lookup_method_type type;
+
+	assert(ev);
+	assert(ev->type == LTTNG_EVENT_USERSPACE_PROBE);
+
+	location = lttng_event_get_userspace_probe_location(ev);
+	if (!location) {
+		ret = -1;
+		goto end;
+	}
+	lookup_method =
+		lttng_userspace_probe_location_get_lookup_method(location);
+	if (!lookup_method) {
+		ret = -1;
+		goto end;
+	}
+	type = lttng_userspace_probe_location_lookup_method_get_type(lookup_method);
+
+	switch (type) {
+	case LTTNG_USERSPACE_PROBE_LOCATION_LOOKUP_METHOD_TYPE_FUNCTION_ELF:
+	{
+		struct lttng_kernel_event_callsite callsite;
+		uint64_t offset;
+		ret = extract_userspace_probe_offset_function_elf(location, session, &offset);
+		if (ret) {
+			ret = LTTNG_ERR_PROBE_LOCATION_INVAL;
+			goto end;
+		}
+
+		callsite.u.uprobe.offset = offset;
+
+		ret = kernctl_add_callsite(fd, &callsite);
+		if (ret) {
+			goto end;
+		}
+		break;
+	}
+	case LTTNG_USERSPACE_PROBE_LOCATION_LOOKUP_METHOD_TYPE_TRACEPOINT_SDT:
+	{
+		int i;
+		uint64_t *offsets = NULL;
+		uint32_t num_offset;
+		struct lttng_kernel_event_callsite callsite;
+		/*
+		 * This call allocates the offsets buffer. This buffer must be freed
+		 * by the caller
+		 */
+		ret = extract_userspace_probe_offset_tracepoint_sdt(location, session,
+				&offsets, &num_offset);
+		if (ret) {
+			ret = LTTNG_ERR_PROBE_LOCATION_INVAL;
+			goto end;
+		}
+		for (i = 0; i < num_offset; i++) {
+			callsite.u.uprobe.offset = offsets[i];
+			ret = kernctl_add_callsite(fd, &callsite);
+			if (ret) {
+				free(offsets);
+				goto end;
+			}
+		}
+		free(offsets);
+		break;
+	}
+	default:
+			ret = LTTNG_ERR_PROBE_LOCATION_INVAL;
+			goto end;
```

Comment:
There is an extra tab on those two lines.

Reply:
done.